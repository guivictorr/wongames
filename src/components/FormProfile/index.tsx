import Button from 'components/Button'
import Heading from 'components/Heading'
import TextField from 'components/TextField'
import Link from 'next/link'
import * as S from './styles'

export type FormProfileProps = {
  username: string
  email: string
}

const FormProfile = ({ email, username }: FormProfileProps) => (
  <>
    <Heading line="bottom" size="small" color="black">
      My profile
    </Heading>

    <S.Form>
      <TextField
        name="username"
        placeholder="Username"
        label="Username"
        initialValue={username}
      />
      <TextField
        name="email"
        type="email"
        placeholder="Email"
        label="Email"
        initialValue={email}
        disabled
      />

      <S.ButtonContainer>
        <Link href={`/forgot-password?email=${email}`} passHref>
          <Button minimal size="medium" as="a">
            Reset Password
          </Button>
        </Link>
        <Button size="large">Save</Button>
      </S.ButtonContainer>
    </S.Form>
  </>
)

export default FormProfile
