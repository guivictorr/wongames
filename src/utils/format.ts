export const formatNumber = (
  value: number,
  options: Intl.NumberFormatOptions
) => {
  return new Intl.NumberFormat('en-US', options).format(value)
}
