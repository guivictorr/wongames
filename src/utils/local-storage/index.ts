const APP_KEY = 'WONGAMES'

export const getStorageItem = (key: string) => {
  if (typeof window === 'undefined') return

  const item = window.localStorage.getItem(`${APP_KEY}_${key}`)

  return JSON.parse(item!)
}

export const setStorageItem = (key: string, value: string | string[]) => {
  if (typeof window === 'undefined') return

  return window.localStorage.setItem(`${APP_KEY}_${key}`, JSON.stringify(value))
}
