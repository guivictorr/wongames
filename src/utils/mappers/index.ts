import { HighlightFragment } from 'graphql/generated/HighlightFragment'
import { QueryGames_games } from 'graphql/generated/QueryGames'
import { QueryHome_banners } from 'graphql/generated/QueryHome'

export const bannerMapper = (banners: QueryHome_banners[]) => {
  return banners.map((banner) => ({
    title: banner.title,
    subtitle: banner.subtitle,
    img: `http://localhost:1337${banner.image?.url}`,
    ...(banner.button && {
      buttonLabel: banner.button.label,
      buttonLink: banner.button.link
    }),
    ...(banner.ribbon && {
      ribbon: banner.ribbon.title,
      ribbonColor: banner.ribbon.color,
      ribbonSize: banner.ribbon.size
    })
  }))
}

export const gameSliderMapper = (
  games: QueryGames_games[] | undefined | null
) => {
  return games
    ? games.map((game) => ({
        id: game.id,
        title: game.name,
        slug: game.slug,
        developer: game.developers[0].name,
        img: `http://localhost:1337${game.cover?.url}`,
        price: game.price
      }))
    : []
}

export const highlightMapper = (
  highlight: HighlightFragment | null | undefined
) => {
  return highlight
    ? {
        title: highlight.title,
        subtitle: highlight.subtitle,
        backgroundImage: `http://localhost:1337${highlight.background?.url}`,
        floatImage: `http://localhost:1337${highlight.floatImage?.url}`,
        buttonLabel: highlight.buttonLabel,
        buttonLink: highlight.buttonLink,
        alignment: highlight.alignment
      }
    : {}
}
