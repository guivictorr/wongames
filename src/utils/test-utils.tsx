import { ReactElement } from 'react'
import { ThemeProvider } from 'styled-components'
import { render, RenderOptions } from '@testing-library/react'
import {
  CartContextDefaultValues,
  CartContext,
  CartContextData
} from 'hooks/useCart'
import theme from 'styles/theme'

type CustomRenderProps = {
  cartProviderProps?: CartContextData
} & Omit<RenderOptions, 'queries'>

const customRender = (
  ui: ReactElement,
  {
    cartProviderProps = CartContextDefaultValues,
    ...renderOptions
  }: CustomRenderProps = {}
) => {
  return render(
    <ThemeProvider theme={theme}>
      <CartContext.Provider value={cartProviderProps}>
        {ui}
      </CartContext.Provider>
    </ThemeProvider>,
    renderOptions
  )
}

export * from '@testing-library/react'
export { customRender as render }
