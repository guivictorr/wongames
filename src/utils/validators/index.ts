import { UsersPermissionsRegisterInput } from 'graphql/generated/globalTypes'
import Joi from 'joi'

const fieldValidators = {
  username: Joi.string().min(5).required(),
  email: Joi.string()
    .email({ tlds: { allow: false } })
    .required(),
  password: Joi.string().required(),
  confirm_password: Joi.string()
    .valid(Joi.ref('password'))
    .required()
    .messages({
      'any.only': 'confirm password does not match with password'
    })
}

export function signUpValidate(values: UsersPermissionsRegisterInput) {
  const schema = Joi.object(fieldValidators)
  return getFieldErrors(schema.validate(values, { abortEarly: false }))
}

export type FieldErrors = {
  [key: string]: string
}

function getFieldErrors(errorObject: Joi.ValidationResult) {
  const errors: FieldErrors = {}

  if (errorObject.error) {
    errorObject.error.details.forEach((detail) => {
      errors[detail.path.join('.')] = detail.message
    })
  }

  return errors
}

type SignInValues = Omit<UsersPermissionsRegisterInput, 'username'>
export function signInValidate(values: SignInValues) {
  const { email, password } = fieldValidators

  const schema = Joi.object<SignInValues>({ email, password })
  const validation = schema.validate(values, { abortEarly: false })
  return getFieldErrors(validation)
}

type ForgotValidateParams = Pick<UsersPermissionsRegisterInput, 'email'>
export function forgotValidate(values: ForgotValidateParams) {
  const { email } = fieldValidators

  const schema = Joi.object({ email })
  const validation = schema.validate(values, { abortEarly: false })
  return getFieldErrors(validation)
}

type ResetValidateParams = {
  password: string
  confirm_password: string
}
export function resetValidate(values: ResetValidateParams) {
  const { password, confirm_password } = fieldValidators

  const schema = Joi.object({ password, confirm_password })
  const validation = schema.validate(values, { abortEarly: false })
  return getFieldErrors(validation)
}
