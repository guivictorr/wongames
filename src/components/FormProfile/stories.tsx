import { Story, Meta } from '@storybook/react'
import FormProfile from '.'

export default {
  title: 'FormProfile',
  component: FormProfile
} as Meta

export const Default: Story = () => <FormProfile />
