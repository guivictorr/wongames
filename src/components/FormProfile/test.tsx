import { screen, render } from 'utils/test-utils'

import FormProfile from '.'

describe('<FormProfile />', () => {
  it('should render the profile form', () => {
    const { container } = render(
      <FormProfile email="email@email.com" username="username" />
    )

    expect(
      screen.getByRole('heading', { name: /my profile/i })
    ).toBeInTheDocument()

    expect(screen.getByRole('textbox', { name: /name/i })).toBeInTheDocument()
    expect(screen.getByRole('textbox', { name: /email/i })).toBeInTheDocument()

    expect(
      screen.getByRole('link', { name: /reset password/i })
    ).toBeInTheDocument()
    expect(screen.getByRole('button', { name: /save/i })).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })
})
