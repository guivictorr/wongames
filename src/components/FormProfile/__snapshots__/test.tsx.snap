// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormProfile /> should render the profile form 1`] = `
.c0 {
  color: #030517;
  position: relative;
  margin-bottom: 3.2rem;
  font-size: 1.6rem;
}

.c0::after {
  content: '';
  position: absolute;
  left: 0;
  bottom: -1rem;
  width: 5rem;
  border-bottom: 0.5rem solid #F231A5;
}

.c0::after {
  width: 3rem;
}

@media (min-width:768px) {

}

@media (min-width:768px) {

}

<h2
  class="c0"
  color="black"
>
  My profile
</h2>
`
